keepassxc 2.6.6 doesn't build on darwin

See Hydra failure:

https://hydra.nixos.org/build/175226618

```
[ 26%] Building CXX object src/CMakeFiles/keepassx_core.dir/gui/styles/base/BaseStyle.cpp.o
/tmp/nix-build-keepassxc-2.6.6.drv-0/source/src/gui/styles/base/BaseStyle.cpp:295:84: error: no member named 'MacOSBigSur' in 'QOperatingSystemVersion'
                if (QOperatingSystemVersion::current() >= QOperatingSystemVersion::MacOSBigSur) {
                                                          ~~~~~~~~~~~~~~~~~~~~~~~~~^
/tmp/nix-build-keepassxc-2.6.6.drv-0/source/src/gui/styles/base/BaseStyle.cpp:302:84: error: no member named 'MacOSBigSur' in 'QOperatingSystemVersion'
                if (QOperatingSystemVersion::current() >= QOperatingSystemVersion::MacOSBigSur) {
                                                          ~~~~~~~~~~~~~~~~~~~~~~~~~^
2 errors generated.
make[2]: *** [src/CMakeFiles/keepassx_core.dir/build.make:1182: src/CMakeFiles/keepassx_core.dir/gui/styles/base/BaseStyle.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:620: src/CMakeFiles/keepassx_core.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
builder for '/nix/store/lxz5mh63aqj2h8kkhf5s6n29jfzv7mhi-keepassxc-2.6.6.drv' failed with exit code 2
```

I could not find anything directly keepassxc-related in the changes on hydra from last successful to first failing build.

CC @jonafato @NixOS/darwin-maintainers 

Also relevant: Version bump to 2.7: https://github.com/NixOS/nixpkgs/pull/166010
A fix for this issue might give a merge conflict there in the worst case, or alleviate the build problems there in the best case.

--- a/pkgs/applications/misc/keepassx/community.nix
+++ b/pkgs/applications/misc/keepassx/community.nix
@@ -1,4 +1,5 @@
-{ lib, stdenv
+{ lib
+, stdenv
 , fetchFromGitHub
 , cmake
 , qttools
@@ -81,8 +82,9 @@ stdenv.mkDerivation rec {
     export LC_ALL="en_US.UTF-8"
     export QT_QPA_PLATFORM=offscreen
     export QT_PLUGIN_PATH="${qtbase.bin}/${qtbase.qtPluginPrefix}"
-    # testcli and testgui are flaky - skip them both
-    make test ARGS+="-E 'testcli|testgui' --output-on-failure"
+    # testcli, testgui and testkdbx4 are flaky - skip them all
+    # testautotype on darwin throws "QWidget: Cannot create a QWidget without QApplication"
+    make test ARGS+="-E 'testcli|testgui${lib.optionalString stdenv.isDarwin "|testautotype|testkdbx4"}' --output-on-failure"
 
     runHook postCheck
   '';
@@ -92,6 +94,8 @@ stdenv.mkDerivation rec {
   dontWrapGApps = true;
   preFixup = ''
     qtWrapperArgs+=("''${gappsWrapperArgs[@]}")
+  '' + lib.optionalString stdenv.isDarwin ''
+    wrapQtApp "$out/Applications/KeePassXC.app/Contents/MacOS/KeePassXC"
   '';
 
   buildInputs = [
@@ -128,6 +132,5 @@ stdenv.mkDerivation rec {
     license = licenses.gpl2Plus;
     maintainers = with maintainers; [ jonafato turion srapenne ];
     platforms = platforms.linux ++ platforms.darwin;
-    broken = stdenv.isDarwin;  # see to https://github.com/NixOS/nixpkgs/issues/172165
   };
 }

